File: sources/academy/webscraping/scraping_basics_javascript/exercises/ikea_dimensions.mjs
import * as cheerio from 'cheerio';

function parseDimensions(text) {
  const words = text.trim().split(' ');
  if (words.at(-1) === 'cm') {
    const dimensions = words.at(-2).split('x');
    if (dimensions.length === 3) {
      return {
        width: dimensions[0],
        depth: dimensions[1],
        height: dimensions[2],
      };
    }
  }
  return { width: null, depth: null, height: null };
}

const url = 'https://www.ikea.com/se/en/cat/storage-solution-systems-46052/';
const response = await fetch(url);

if (!response.ok) {
  throw new Error(`HTTP ${response.status}`);
}

const html = await response.text();
const $ = cheerio.load(html);

for (const element of $('.plp-mastercard').toArray()) {
  const $productCard = $(element);

  const $title = $productCard.find('.plp-price-module__product-name');
  const title = $title.text().trim();

  const descriptionText = $productCard.find('.plp-text').text();
  const dimensions = parseDimensions(descriptionText);

  console.log(`${title} | w ${dimensions.width} | d ${dimensions.depth} | h ${dimensions.height}`);
}

File: sources/academy/webscraping/scraping_basics_python/exercises/ikea_dimensions.py
import httpx
from bs4 import BeautifulSoup


def parse_dimensions(text):
    words = text.strip().split()
    if words and words[-1] == "cm":
        dimensions = words[-2].split("x")
        if len(dimensions) == 3:
            return dimensions
    return None


url = "https://www.ikea.com/se/en/cat/jonaxel-system-45730/"
response = httpx.get(url)
response.raise_for_status()

soup = BeautifulSoup(response.text, "html.parser")

for product in soup.select(".plp-mastercard"):
    description_text = product.select_one(".plp-text").text
    if dimensions := parse_dimensions(description_text):
        price = product.select_one(".plp-price__integer").text.replace(" ", "").strip()
        print(f"{dimensions[0]} | {dimensions[1]} | {dimensions[2]} ... {price} SEK")
